Rock/Communication/Transport/Firebase.cs

```diff
@@ -0,0 +1,244 @@
+    class Firebase : TransportComponent
+    {
+        /// <summary>
+        /// Sends the specified communication.
+        /// </summary>
+        /// <param name="communication">The communication.</param>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override void Send( Rock.Model.Communication communication )
+        {
+            var rockContext = new RockContext();
+
+            // Requery the Communication
+            communication = new CommunicationService( rockContext ).Get( communication.Id );
+
+            if ( communication != null &&
+                communication.Status == Model.CommunicationStatus.Approved &&
+                communication.HasPendingRecipients(rockContext) &&
+                ( !communication.FutureSendDateTime.HasValue || communication.FutureSendDateTime.Value.CompareTo( RockDateTime.Now ) <= 0 ) )
+            {
+
+                string senderId = GetAttributeValue( "SenderId" );
+                string serverKey = GetAttributeValue( "ServerKey" );
+                var sender = new Sender(serverKey);
+
+                var personEntityTypeId = EntityTypeCache.Read( "Rock.Model.Person" ).Id;
+                var communicationEntityTypeId = EntityTypeCache.Read( "Rock.Model.Communication" ).Id;
+                var communicationCategoryId = CategoryCache.Read( Rock.SystemGuid.Category.HISTORY_PERSON_COMMUNICATIONS.AsGuid(), rockContext ).Id;
+
+                var mergeFields = Rock.Lava.LavaHelper.GetCommonMergeFields( null );
+
+                // get message template
+                string message = communication.GetMediumDataValue( "Message" );
+
+                // get message title
+                string title = communication.GetMediumDataValue("Title");
+
+                // get sound preference
+                string sound = communication.GetMediumDataValue("Sound");
+
+                // convert any special microsoft word characters to normal chars so they don't look funny (for example "Hey â€œdouble-quotesâ€ from â€˜single quoteâ€™")
+                message = message.ReplaceWordChars();
+
+                bool recipientFound = true;
+                while ( recipientFound )
+                {
+                    var loopContext = new RockContext();
+                    var recipient = Rock.Model.Communication.GetNextPending( communication.Id, loopContext );
+                    var historyService = new HistoryService( loopContext );
+                    if ( recipient != null )
+                    {
+                        try
+                        {
+                            var service = new PersonalDeviceService( rockContext );
+                            int personAlias = recipient.PersonAliasId;
+
+                            List<string> devices = service.Queryable()
+                                .Where(p => p.PersonAliasId == personAlias && p.NotificationsEnabled)
```

Comment:
Not a big deal, but if you are changing other things, this same `.Where` clause is used in several places. You might want to consider creating a service partial class and a method for this. i.e. 
```
        public IQueryable<PersonalDevice> GetByPersonAliasId( int personAliasId )
        {
            return Queryable().Where( p => p.PersonAliasId == personAliasId && p.NotificationsEnabled );
        }
```